### PR TITLE
Enable auto merging of tablets

### LINKED ISSUE #5014: Add identifier to split points for automated vs user created split points
**Is your feature request related to a problem? Please describe.**
When split points are automatically created due to tablet sizes, these split points persist until a user runs a merge. 
If a merge is run automatically, then it might remove split points that were purposely set by the user.

**Describe the solution you'd like**
Split points created automatically by accumulo should be identified as such so they can easily be removed.

### PR DESCRIPTION
**Pull request by cshannon**
This change adds support for periodically scanning tables to find ranges of tables that can be automatically merged. The thread that runs uses the recently added TabletMergeabilty column as well as total file counts/sizes to compute if a range can be merged. By default, tablets can be merged if the total size is less than 25% of the split threshold so we don't immediately merge only to split again.

There is a new thread in the Manager class that performs the computation and will submit fate jobs for merging on tablet ranges. There is a new fate operation during merge that will validate the range is still ok to merge if the merge was submitted as a system merge.

This closes #5014

I am making this a draft because while it's feature complete, I still think there's some more work to do on the tests and tweaks to make, but wanted to get some feedback/suggestions.

#### Outstanding issues/note/questions:

1. **For user created tables, the default tablet was previously set to `never` merge, but I think we should change this to always so i switched the default tablet to be a mergeability of `always` in this PR.** The reason is we can split the default tablet, so we should be able to merge other tablets back to it. It will never go away because the end row is null. One reason why I did this is there is no way in the API right now to change the mergeability for the default tablet (end row is null which you can't put into a map as a key) so I figure we should just make it always mergeable and any other splits would not be mergeable by default for user created tablets. This would be easy to set back to never by default if desired, but if keep the default tablet as never then we can never merge back to a single tablet again.
2. The default tables (metadata, fate, scanref) are configured with their initital tablet(s) to have a mergeability setting of `never`. Should we change this? I am thinking I probably want to switch the default tablet to be mergeable just like I did for user tables for consistency. However, pre-splitting is ever done the tablets would need to be made sure to be marked as never merge so they don't get merged away due to a size of 0.
3. What thread pool should we use for `FindMergeableRangeTask`? What interval by default? For now I put in a todo marker and just used the default context scheduler and set a run time of every 24 hours.
4. As mentioned, the tests could definitely be improved with more edge cases and complexity. The `TabletMergeabilityIT` especially only has 4 tests for now to demonstrate but there's a lot more tests we could write to verify the checks work like max files, sizes, etc work and we get the exepcted merges done.

### CODE REVIEW

**Comment by cshannon** [comment]
Looks like there's a couple tests to fix which I can do when I get a chance, they are boken cause the PR changes the default tablet mergeability setting to always instead of never as mentioned in the description.

```
org.apache.accumulo.test.functional.CreateInitialSplitsIT.testCreateInitialSplitsWithMergeability
org.apache.accumulo.test.functional.SplitIT.tabletShouldSplit
```

File: server/manager/src/main/java/org/apache/accumulo/manager/Manager.java [hunk 1]
```diff
@@ -1292,6 +1293,13 @@ boolean canSuspendTablets() {
     ThreadPools.watchCriticalScheduledTask(context.getScheduledExecutor()
         .scheduleWithFixedDelay(() -> ScanServerMetadataEntries.clean(context), 10, 10, MINUTES));
 
+    // TODO - create new threadpool?
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/Manager.java:1296]: I assume that this cannot go into the TabletGroupWatcher because it needs to look at more than one Tablet? I think using the general scheduled executor threadpool should be fine. However, we may want to consider bumping the default value of `GENERAL_THREADPOOL_SIZE` from `1` to something larger. Probably good for a separate PR.

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        return true;
+      }
+      return false;
+    }
+
+    private boolean validate(TabletMetadata tm) {
+      if (tabletCount > 0) {
+        // If this is not the first tablet, then verify its prevEndRow matches
+        // the last endRow tracked, the server filter will skip tablets marked as never
+        if (!tm.getPrevEndRow().equals(endRow)) {
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:181]: `tm.getPrevEndRow()` could be null? `endRow` could be null?

**Comment by dlmarion** [comment]
Is there work to do in the upgrade code, or has that already been done?

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java (hunk 2 above)
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:181]: End row can definitely be null, prevEndRow _should_ not be null because the `if (tabletCount > 0) ` means it's at least the second tablet seen. However it would probably be good to add a null check just in case. If it is null we should throw an IllegalStateException probably.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java [hunk 3]
```diff
@@ -357,4 +358,12 @@ boolean testClassLoad(final String className, final String asTypeName)
    * @since 2.1.0
    */
   InstanceId getInstanceId();
+
+  /**
+   * Return the current manager time
```
[the remark this answers is not in the document]
Comment (cshannon) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:363]: Manager time is the SteadyTime that is tracked by the manager and stored in ZK for the instance so even if the Manager changes it won't go backwards as it's persisted across restarts

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java [hunk 4]
```diff
@@ -904,6 +911,8 @@ public enum Property {
   TABLE_ONDEMAND_UNLOADER("tserver.ondemand.tablet.unloader",
       "org.apache.accumulo.core.spi.ondemand.DefaultOnDemandTabletUnloader", PropertyType.CLASSNAME,
       "The class that will be used to determine which on-demand Tablets to unload.", "4.0.0"),
+  TABLE_MAX_MERGEABILITY_THRESHOLD("table.mergeability.threshold", ".25", PropertyType.FRACTION,
+      "A tablet is mergeable until it reaches this percentage of the split threshold.", "1.3.5"),
```
[the remark this answers is not in the document]
Comment (cshannon) [reply core/src/main/java/org/apache/accumulo/core/conf/Property.java:915]: Copy and paste error, will fix that.

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java [hunk 5]
```diff
@@ -28,6 +28,8 @@
 import java.util.Objects;
 import java.util.function.Predicate;
 
+import javax.print.DocFlavor.STRING;
```
[the remark this answers is not in the document]
Comment (cshannon) [reply core/src/main/java/org/apache/accumulo/core/conf/Property.java:31]: I have no idea where that came from, I'm surprised the Maven build didn't remove it or error.

File: server/manager/src/main/java/org/apache/accumulo/manager/Manager.java (hunk 1 above)
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/Manager.java:1296]: Yes, TGW doesn't work in this case because because the task needs to look at multiple tablets together and not just one tablet at a time. Also, the task likely doesn't need to run very often to look for mergeable ranges where as TGW runs frequently

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+    }
+
+    log.debug("Table {} found {} tablets that can be merged for table", table.getValue(),
+        range.tabletCount);
+
+    TableId tableId = table.getKey();
+    String tableName = table.getValue();
+
+    range.startRow = Optional.ofNullable(range.startRow).map(Text::new).orElse(new Text(""));
+    range.endRow = Optional.ofNullable(range.endRow).map(Text::new).orElse(new Text(""));
```
[the remark this answers is not in the document]
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:129]: Yeah this should probably be re-worked to not change the original fields of the input variable, this is a holdover from when I was testing and trying different things and got missed so I will refactor and also get rid of the extra object creation as suggested. The start/end row need to be empty text and not null when submitted which is why it is being changed.

The code was modeled after the code [here](https://github.com/apache/accumulo/blob/121758a40e35df5589e53d9cfecdd95f7705cc56/server/manager/src/main/java/org/apache/accumulo/manager/FateServiceHandler.java#L456-L490) for handling merge in FateServiceHandler.

**Comment by cshannon** [comment]
> Is there work to do in the upgrade code, or has that already been done?

There is still upgrade work to do, we should go through and mark all existing tablets as "never" merge on upgrade. If I recall the default will be never if the column is absent but I think it's better to be explicit and mark the tablets.

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java]: Fixed in https://github.com/apache/accumulo/pull/5353/commits/81f2c3723c5db9ae14826c9f2a8d1c653ad32a9c

File: test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+      // Ingest data so tablet will split
+      VerifyParams params = new VerifyParams(getClientProps(), tableName, 50_000);
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
+      // Wait for table to split, should be more than 10 tablets
+      Wait.waitFor(() -> c.tableOperations().listSplits(tableName).size() > 10, 10000, 200);
+
+      // Delete all the data
+      c.tableOperations().deleteRows(tableName, null, null);
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java:166]: deleteRows also merges tablets I think, so the splits may not be going away because of auto merge here.  Could delete all the data and then compact to process all the delete markers.


```suggestion
       var bd = c.createBatchDeleter(tableName, Authorizations.EMPTY, 1);
      bd.setRanges(List.of(new Range()));
      bd.delete();
      c.tableOperations().compact(tableName, new CompactionConfig().setFlush(true));
```

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/VerifyMergeability.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+    try (var tablets = env.getContext().getAmple().readTablets().forTable(data.tableId)
+        .overlapping(range.prevEndRow(), range.endRow()).fetch(FILES, MERGEABILITY)
+        .checkConsistency().build()) {
+
+      long totalSize = 0;
+      long totalFiles = 0;
+      int totalUnMergeable = 0;
+
+      for (var tabletMetadata : tablets) {
```
[the remark this answers is not in the document]
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/VerifyMergeability.java:73]: I thought the same thing, I have/had been trying to figure out a good way to share the same logic between both but it's just different enough there wasn't really a good way to share the code but i'll keep thinking about it.

[the remark this answers is not in the document]
Comment (keith-turner) [reply server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/VerifyMergeability.java:73]: Wondering if MergeableRange.add() returned an enum instead of a boolean if that could work.  The enum would indicate if added or if not added then why.

File: test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java (hunk 7 above)
Comment (cshannon) [reply test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java:166]: Ah that is a good point, I forgot that deleteRows merges away empty tablets already so I will make that change.

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+    String startRowStr = StringUtils.defaultIfBlank(startRow.toString(), "-inf");
+    String endRowStr = StringUtils.defaultIfBlank(endRow.toString(), "+inf");
+    log.debug("FindMergeableRangeTask: Creating merge op: {} from startRow: {} to endRow: {}",
+        tableId, startRowStr, endRowStr);
+    var fateId = manager.fate(type).startTransaction();
+    String goalMessage = TableOperation.MERGE + " Merge table " + tableName + "(" + tableId
+        + ") splits from " + startRowStr + " to " + endRowStr;
+
+    manager.fate(type).seedTransaction(FateOperation.SYSTEM_MERGE, fateId,
```
[the remark this answers is not in the document]
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:139]: This is a really good idea, I will make this change for sure.

File: test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java (hunk 7 above)
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java]: I was not sure,  I had to look at the impl.  The javadoc does not mention that delete rows will change the tables split point, its should.  Actually it does not have to change the splits, it could leave the splits alone and just drop files from tablets.  I think it does that because it was implemented using the merge tablet code.  Also in 2.1 it may add and remove splits, not sure but in 3.1 think it no longer add splits to the table and only removes.

Comment (cshannon) [reply test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java]: If I recall, I believe the reason why we use the merge code to merge away empty tablets is to preserve the behavior of previous versions of accumulo that deleted tablets. 

In older versions when we ran delete rows, the code would run a chop compaction on the tablets at the boundaries and then it could just delete the all the tablets in between the delete range.

With the addition of no-chop merges, we now have the ability to just fence files and avoid chop compactions. So now we fence the tablets on the boundaries, and can delete the metadata in between and merge away the empty tablets.

The end result in both cases is those tablets are gone, so if we do decide to keep the tablets going forward that would be a different behavior than before.

Comment (cshannon) [reply test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java]: Should we take advantage of the new TabletMergeability column? Maybe we should change deleteRows to only merge away the empty tablets if they are marked as mergeable?

Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java]: Opened #5371 based on this disucssion.

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java [hunk 10]
```diff
@@ -132,18 +133,17 @@ void submit(MergeableRange range, FateInstanceType type, Entry<TableId,String> t
     String endRowStr = StringUtils.defaultIfBlank(endRow.toString(), "+inf");
     log.debug("FindMergeableRangeTask: Creating merge op: {} from startRow: {} to endRow: {}",
         tableId, startRowStr, endRowStr);
-    var fateId = manager.fate(type).startTransaction();
-    String goalMessage = TableOperation.MERGE + " Merge table " + tableName + "(" + tableId
-        + ") splits from " + startRowStr + " to " + endRowStr;
+    var fateKey = FateKey.forMerge(new KeyExtent(tableId, range.endRow, range.startRow));
 
-    manager.fate(type).seedTransaction(FateOperation.SYSTEM_MERGE, fateId,
+    manager.fate(type).seedTransaction(FateOperation.SYSTEM_MERGE, fateKey,
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:138]: This will wait for the data to be written to the walog of the fate table for each operation seeded.  W/ the changes in #5160 this is another place where waiting on the walog per operation can be avoided.

File: test/src/main/java/org/apache/accumulo/test/fate/FateStoreIT.java [hunk 11]
```diff
@@ -683,8 +689,12 @@ protected void testListFateKeys(FateStore<TestEnv> store, ServerContext sctx) th
     var fateKey3 = FateKey.forCompactionCommit(cid1);
     var fateKey4 = FateKey.forCompactionCommit(cid2);
 
+    // use one overlapping extent and one different
+    var fateKey5 = FateKey.forMerge(extent1);
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/fate/FateStoreIT.java:693]: Not a change for this PR.  Was looking around to see if FateKey had any unit test that should be updated and did not see any.  Could be useful to add some unit test to check equals, hashcode, serder, etc in another PR.

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/VerifyMergeability.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/VerifyMergeability.java:69]: I refactored things to share MergeableRange between the task and verify in 3c84e8553c9f0a9e21b959d480fd798be284946c

File: test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java [hunk 12]
```diff
@@ -183,6 +184,48 @@ public void testSplitAndMergeAll() throws Exception {
...
+      // Set the split threshold back to the default of 5 MB. There's not a lot of data so normally
+      // we could merge back to 1 tablet, but the threshold is too low at 1% so it should not merge
+      // yet.
+      c.tableOperations().setProperty(tableName, Property.TABLE_SPLIT_THRESHOLD.getKey(), "5m");
+
+      // Should not merge so make sure it throws IllegalStateException
+      assertThrows(IllegalStateException.class,
+          () -> Wait.waitFor(() -> hasExactTablets(getCluster().getServerContext(), tableId,
+              Set.of(new KeyExtent(tableId, null, null))), 5000, 500));
+
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java:218]: Could call hasExactTablets again after the waitFor, this helps confirm the IllegalStateException was thrown for the expected reason.  Assuming the test expects the IllegalSE to be thrown on timeout?

**Comment by cshannon** [comment]
I ran the full IT build and it passed so this is ready for another review.

File: test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java [hunk 13]
```diff
@@ -0,0 +1,406 @@
...
+          5000, 500);
+
+      // Ingest data so tablet will split, each tablet will have several files because
+      // of the flush setting
+      VerifyParams params = new VerifyParams(getClientProps(), tableName, 5_000);
+      params.startRow = 0;
+      params.flushAfterRows = 100;
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java:362]: Could add a check of the number of files here.  Maybe it should do a waitFor instead of assertTrue though.

```
assertTrue(FileMetadataUtil.countFiles(getCluster().getServerContext(), tableName) > 3);
```

File: test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java [hunk 14]
```diff
@@ -0,0 +1,406 @@
...
+              Set.of(new KeyExtent(tableId, null, null))), 5000, 500));
+      // Make sure tablets is still 10, not merged
+      assertEquals(10, countTablets(getCluster().getServerContext(), tableName, tm -> true));
+
+      // Set max merge file count back to default of 10k
+      c.tableOperations().setProperty(tableName, Property.TABLE_MERGE_FILE_MAX.getKey(), "10000");
+
+      // Should merge back to 1 tablet
+      Wait.waitFor(() -> hasExactTablets(getCluster().getServerContext(), tableId,
+          Set.of(new KeyExtent(tableId, null, null))), 10000, 200);
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java:382]: Can re-verify the data after merge.  There are a few other test were the data could be verified again after the merges.

```suggestion
          Set.of(new KeyExtent(tableId, null, null))), 10000, 200);
          
      VerifyIngest.verifyIngest(c, params);
```

File: server/manager/src/main/java/org/apache/accumulo/manager/Manager.java [hunk 15]
```diff
@@ -1309,6 +1310,13 @@ boolean canSuspendTablets() {
...
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/Manager.java:1313]: Need to remove this TODO

**Review by keith-turner — APPROVED** [review]
Everything looks good.

> What thread pool should we use for FindMergeableRangeTask? What interval by default? For now I put in a todo marker and just used the default context scheduler and set a run time of every 24 hours.

Not something for this PR and maybe its never needed, but in addition to the periodic scan could also have scans triggered by events.   Like if a `table.mergeability.threshold` is changed for a table or the mergability of some tablets are changed then trigger a scan.  

> As mentioned, the tests could definitely be improved with more edge cases and complexity. The TabletMergeabilityIT especially only has 4 tests for now to demonstrate but there's a lot more tests we could write to verify the checks work like max files, sizes, etc work and we get the exepcted merges done.

The test look good, is there anything else you were going to add?  One other that would be nice is ensure the metadata table can automerge, this feature will be really nice to use w/ that table.

**Comment by cshannon** [comment]
> Not something for this PR and maybe its never needed, but in addition to the periodic scan could also have scans triggered by events. Like if a `table.mergeability.threshold` is changed for a table or the mergability of some tablets are changed then trigger a scan.

This is something that could be nice to add later if we need it, it would not be too hard so we can wait and see.

> The test look good, is there anything else you were going to add? One other that would be nice is ensure the metadata table can automerge, this feature will be really nice to use w/ that table.

I added all the tests I wanted to, in regards to to automerge for metadata that was something I forgot to bring up and wanted to ask about.

Originally the default tablet for normal user table creation was set to "never "merge but that was [changed](https://github.com/apache/accumulo/pull/5353/files#diff-223656295941b1bcacbd49a1d0067f5f2c90af24c461f99ef5e490cdef21d2f0R262) in this PR to be "always" for user created tablets to allow merging back to one tablet if other splits are eligible for merging.

However, the initial system tables still have their default tablets [marked](https://github.com/apache/accumulo/blob/0c79e44e2df29f8f9d8617fb698167a60c0512bb/server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java#L98) as never.

One reason why was I was not sure if we wanted the system tables to be eligible to be auto merged or not. I assume by default it would be fine, if someone wants to manually split the metadata table, fate table, etc and keep tablets they can just mark those tablets as "never" merge like any other split if they want.

But the metadata tablet is a bit special though as it is [initialized](https://github.com/apache/accumulo/blob/0c79e44e2df29f8f9d8617fb698167a60c0512bb/server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java#L163-L169) with 2 tablets and not 1 so I would assume we would never want to merge those both back together? And if so I was not sure how/if we should handle auto mergeability. Depending on what is decided here I can add another test.

**Comment by keith-turner** [comment]
> But the metadata tablet is a bit special though as it is [initialized](https://github.com/apache/accumulo/blob/0c79e44e2df29f8f9d8617fb698167a60c0512bb/server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java#L163-L169) with 2 tablets and not 1 so I would assume we would never want to merge those both back together? And if so I was not sure how/if we should handle auto mergeability. Depending on what is decided here I can add another test.

IMO that split added by the initialization code should have a mergability of never set by the init code.   The split is there because it separates two very different types of data the metadata table stores.

**Comment by cshannon** [comment]
> IMO that split added by the initialization code should have a mergability of never set by the init code. The split is there because it separates two very different types of data the metadata table stores.

The latest updates set system tablets with mergeability settings to always, except for the metadata split point tablet of never so it doesn't go away on merge. On upgrade it defaults to never for existing systems so users don't have unexpected behavior. If that needed to be changed in the future it could (But would need to look up the steady time from the manager using the client API to set a mergeability setting on upgrade). Tests have been added for the system tables (metadata, scanref, fate)

File: server/manager/src/main/java/org/apache/accumulo/manager/Manager.java (hunk 1 above)
Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/Manager.java]: I opened https://github.com/apache/accumulo/issues/5405 for this as a possible follow on for bumping the thread pool size

File: server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java [hunk 16]
```diff
@@ -0,0 +1,293 @@
...
+            });
+          }
+
+          // Try and submit any outstanding mergeable tablets
+          submit(current, type, table, namespaceId);
+        }
+
+      } catch (Exception e) {
+        log.error("Failed to generate system merges for {}", tableName, e);
+      }
```
Comment (ctubbsii) [inline server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:134]: @cshannon This completely swallows any exception, even new ones added to the code later. Are you sure that is what is intended here?

It seems to me that, at the very least, this could be narrowed to the current exceptions, so it doesn't catch unintended things that get added later:

```java
      } catch (TableNotFoundException | RuntimeException e) {
        log.error("Failed to generate system merges for {}", tableName, e);
      }
```

Comment (cshannon) [reply server/manager/src/main/java/org/apache/accumulo/manager/merge/FindMergeableRangeTask.java:134]: I pushed an update to main with this change.
